src/producer.rs

```diff
@@ -479,6 +484,7 @@ impl<T> Producer<T> {
             Ok(false) => {
                 let response = self.0.client.delete_publisher(self.0.producer_id).await?;
                 if response.is_ok() {
+                    self.0.client.close().await?;
```

Comment:
@DanielePalaia @wolf4ood we need to close the TCP client here. but I am not sure that this is the right way.